File: 4주차/이연걸/배열복원하기.py
'''
h, w, x, y = map(int, input().split())

b = []
a = [[0] * (w) for _ in range(h)]
for i in range(h + x):
    b.append(list(map(int, input().split())))

for i in range(x):
    for j in range(w):
        a[i][j] = b[i][j]

for i in range(h, len(b)):
    for j in range(y, len(b[0])):
        a[i - x][j - y] = b[i][j]

for i in range(h):
    for j in range(w):
        print(a[i][j], end=' ')
    print()

1  2  3  4
5  6  7  8
9  10 11 12
13 14 15 16

0  0  0  0  0  0 
0  0  0  0  0  0
0  0  1  2  3  4
0  0  5  6  7  8
0  0  9  10 11 12
0  0  13 14 15 16

4 4 2 2
1  2  3  4  0  0 
5  6  7  8  0  0 
9  10 12 14 3  4
13 14 20 22 7  8
0  0  9  10 11 12
0  0  13 14 15 16
'''

h, w, x, y = map(int, input().split())
a, b = [[0] * w for _ in range(h)], []

for _ in range(h + x):
    b.append(list(map(int, input().split())))

for i in range(h):
    for j in range(w):
        a[i][j] = b[i][j]

for i in range(x, h):
    for j in range(y, w):
        a[i][j] = b[i][j] - a[i-x][j-y]

for i in range(h):
    for j in range(w):
        print(a[i][j], end=' ')
    print()

File: 4주차/임유정/배열 복원하기.js
// https://www.acmicpc.net/problem/16967

// 입력 처리
const file = "/dev/stdin";

let input = require("fs").readFileSync(file).toString().trim().split("\n");
const [A, B, X, Y] = input
  .shift()
  .split(" ")
  .map((e) => parseInt(e));
// 이차배열 input정리
const arrB = input.map((row) => row.split(" ").map((col) => Number(col)));
const arrA = Array.from({ length: A }, () =>
  Array.from({ length: B }, () => 0)
);

// 겹치지않는 왼쪽 위 부분
for (let i = 0; i < X; i++) {
  for (let j = 0; j < B; j++) {
    arrA[i][j] = arrB[i][j];
  }
}
// 안 겹치는 곳 왼쪽
for (let i = X; i < A; i++) {
  for (let j = 0; j < Y; j++) {
    arrA[i][j] = arrB[i][j];
  }
}
//겹치는 부분
for (let i = X; i < A; i++) {
  for (let j = Y; j < B; j++) {
    arrA[i][j] = arrB[i][j] - arrA[i - X][j - Y];
  }
}

arrA.forEach((row) => console.log(row.join(" ")));
